boa/src/object/gcobject.rs

```diff
@@ -232,14 +238,22 @@ impl JsObject {
                                     && !body.function_declared_names().contains("arguments")))
                         {
                             // Add arguments object
-                            let arguments_obj = create_unmapped_arguments_object(args, context)?;
+                            let arguments_obj =
+                                if context.strict() || body.strict() || !is_simple_parameter_list {
```

Comment:
@raskad Do we need to check both `context` and `body` for strictness? If we do, can we pass the global strictness into all functions? Because that would remove the context check

Reply:
I think in theory we could pass the global strictness in all functions. But my proposal would be to wait with the strictness refactor until we have some more errors that depend on strictness implemented. Then we can be somewhat sure that we do not break anything and we know how the apis to manage strictness should look.